src/vtquery.cpp

```diff
@@ -99,58 +55,203 @@ struct Worker : Nan::AsyncWorker
     }
 
     std::string result_;
-    const bool louder_;
 };
 
-// vtquery is a "standalone function" because it's not a class.
-// If this function was not defined within a namespace ("VectorTileQuery"
-// specified above), it would be in the global scope.
 NAN_METHOD(vtquery)
 {
-
-    bool louder = false;
-
-    // Check second argument, should be a 'callback' function.
-    // This allows us to set the callback so we can use it to return errors
-    // instead of throwing.
-    // Also, "info" comes from the NAN_METHOD macro, which returns differently
-    // according to the version of node
-    if (!info[1]->IsFunction())
+    // validate callback function
+    v8::Local<v8::Value> callback_val = info[info.Length() - 1];
+    if (!callback_val->IsFunction())
     {
-        return Nan::ThrowTypeError("second arg 'callback' must be a function");
+        Nan::ThrowError("last argument must be a callback function");
+        return;
     }
-    v8::Local<v8::Function> callback = info[1].As<v8::Function>();
+    v8::Local<v8::Function> callback = callback_val.As<v8::Function>();
 
-    // Check first argument, should be an 'options' object
-    if (!info[0]->IsObject())
-    {
-        return utils::CallbackError("first arg 'options' must be an object",
-                                    callback);
+    // validate buffers
+      // must have `buffer` buffer
+      // must have `z` int >= 0
+      // must have `x` int >= 0
+      // must have `y` int >= 0
+    if (!info[0]->IsArray())
+        return utils::CallbackError("first arg 'buffers' must be an array of objects", callback);
+
+    v8::Local<v8::Array> buffers_arr_val = info[0].As<v8::Array>();
+    unsigned num_buffers = buffers_arr_val->Length();
+
+    if (num_buffers <= 0)
+      return utils::CallbackError("'buffers' array must be of length greater than 0", callback);
+
+    // std::vector<> - some sort of storage mechanism for buffers and their zxy values here
```

Comment:
@springmeyer any idea what a good data structure would be here? A vector that contains maps of `buffer, x, y, z` values? Like:

```
std::vector<std::map<...>>
```

❓

Reply:
Or perhaps we create a custom struct that has these values defined, so we could do a vector of structs with the proper values?